viewer/src/Main.rs

```diff
@@ -280,41 +278,76 @@ impl App {
                     draw_len.try_into().unwrap(),
                 )?;
             }
+
             self.canvas.present();
-            thread::sleep(Duration::new(0, 1_000_000_000u32 / 30));
+
+            thread::sleep(Duration::new(0, (1_000_000_000u32 - loop_time.elapsed().as_millis() as u32) / 30));
         }
 
         Ok(())
     }
 }
 
-pub fn load_args() -> Vec<Span> {
+pub fn load_args(mut args: Vec<String>) -> Vec<Span> {
     let mut spans = vec![];
-    let args: Vec<_> = env::args().collect();
     match args.len() {
         1 => {panic!("Command line arguments were not provided. Format: (file path) (span range start) (span range stop) (tag range start) (tag range stop).")},
+        2 => {
+            args.push(0.to_string());
+            spans = load_args(args);
+        },
+        3 => {
+            args.push(u64::MAX.to_string());
+            spans = load_args(args);
+        },
+        4 => {
+            args.push(0.to_string());
+            spans = load_args(args);
+        },
+        5 => {
+            println!("One or more arguments not provided. Running with defaults for missing values.");
+            args.push(u64::MAX.to_string());
+            spans = load_args(args);
+        },
```

Comment:
OK for now, but I definitely think if you add / change args any more than this, you should reach for https://docs.rs/clap/latest/clap/